src/main.rs

```diff
@@ -225,21 +253,20 @@ async fn run(opts: Opts) -> Result<(), BoxError> {
             | Ok(DebouncedEvent::Remove(path))
             | Ok(DebouncedEvent::Write(path)) => {
                 if *path == wasm {
-                    log::debug!("notify: {:?}", event.unwrap());
-                    if let Ok(module) = load_module(&engine, &wasm) {
-                        log::debug!("replacing module");
+                    log::trace!("notify: {:?}", event.unwrap());
+                    if let Ok(module) = load_module(&engine, &wasm, false) {
                         state.write().unwrap().module = module;
                     }
                 }
             }
-            Err(e) => log::debug!("watch error: {:?}", e),
+            Err(e) => {
+                log::trace!("watch error: {:?}", e);
+                break;
```

Comment:
I noticed that when the tx ref is dropped this thread kept spinning in an endless loop with a recv error which is expected but not a useful use of resources. Instead this now breaks on the first recv error as its a non recoverable error.